src/event/validation.ts

```diff
@@ -82,7 +85,8 @@ export const asBuffer = (value: string | Buffer | Uint32Array): Buffer =>
         throw new TypeError("is not buffer or a valid binary");
       })();
 
-export const asBase64 = (value: string | Buffer | Uint32Array): string => asBuffer(value).toString("base64");
+export const asBase64 = 
+(value: string | Buffer | Uint32Array | Uint16Array | Uint8Array): string => asBuffer(value).toString("base64");
```

Comment:
Is this type exhaustive of all types? Maybe we can use a separate typed array `type` here.

https://stackoverflow.com/a/51681541